src/shared/defaultAwsContextCommands.ts

```diff
@@ -81,19 +81,19 @@ export class DefaultAWSContextCommands {
 
     public async onCommandLogin() {
         const profileName = await this.getProfileNameFromUser()
-        let successfulLogin = false
-        if (profileName) {
-            successfulLogin = await UserCredentialsUtils.addUserDataToContext(profileName, this._awsContext)
-            if (successfulLogin) {
-                this.refresh()
-                await this.checkExplorerForDefaultRegion(profileName)
-            } else {
-                await this.onCommandLogout()
-                // tslint:disable-next-line: no-floating-promises
-                UserCredentialsUtils.badUserCredentialPrompt(profileName)
-            }
+        if (!profileName) {
+            // user clicked away from quick pick or entered nothing
+            return
+        }
+        const successfulLogin = await UserCredentialsUtils.addUserDataToContext(profileName, this._awsContext)
+        if (successfulLogin) {
+            this.refresh()
+            await this.checkExplorerForDefaultRegion(profileName)
+        } else {
+            await this.onCommandLogout()
+            // tslint:disable-next-line: no-floating-promises
+            UserCredentialsUtils.notifyUserCredentialsAreBad(profileName)
```

Comment:
Nit: It's fine to await here, because `onCommandLogin` is already fire-and-forget. Same comment below.